pkg/cache/shares.go

```diff
@@ -160,11 +165,42 @@ func UpdateSharedSecret(share *sharev1alpha1.SharedSecret) {
 	br := share.Spec.SecretRef
 	key := BuildKey(br.Namespace, br.Name)
 	secretsWithShare.Delete(key)
-	AddSharedSecret(share)
+	return
+}
+
+func NamespacesWithSharedConfigMaps() []string {
+	namespacesMap := map[string]struct{}{}
+	namespaces := []string{}
+	list, err := client.GetListers().SharedConfigMaps.List(labels.Everything())
+	if err == nil {
+		for _, scm := range list {
+			namespacesMap[scm.Spec.ConfigMapRef.Namespace] = struct{}{}
+		}
+	}
+	for ns := range namespacesMap {
+		namespaces = append(namespaces, ns)
+	}
```

Comment:
I would contend that it would be better to refactor the `PruneSecretInformers` function to take a `map[string]struct{}` as a parameter, and get rid of translating the `namespacesMap` to a `[]string` here and returning it. It would save some processor cycles and memory.

Reply:
agreed ... will update

Reply:
updated @coreydaley